gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceExtractor.java

```diff
@@ -896,6 +927,43 @@ private void reinitializeBufferedReader() throws IOException, AsyncApiException
     }
   }
 
+  private List<JsonElement> fetchPkChunkingResultSetWithRetry(
```

Comment:
So, bulkConnection.getQueryResultStream does not have Retry? 

If we need retry, we should some backoff to avoid immediate attempts as consecutive retries without waiting might be fruitless? In this case, we can leverage `Retryer`, checkout `WriterUtils.mkdirsWithRecursivePermissionWithRetry` for an example. It's straightforward to make the change.

Reply:
You are right! There is no wait! I copied the retry logic from **fetchResultBatchWithRetry**  and didn't look into it.
I have another thought - Since the gobblin would re-execute the workUnit again anyway when the workUnit fails, we can just remove the retry. 
how do you think of?

Reply:
I see. It's an existing logic. Then that's fine.